packages/gatsby-theme-store/src/components/Auth/Providers/ExternalProvider/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { Alert, Box, Button, Center, Flex, Spinner } from '@vtex/store-ui'
+import type { AuthProviderComponentProps } from '@vtex/store-ui'
+import { useIntl } from '@vtex/gatsby-plugin-i18n'
+import React, { useEffect, useState } from 'react'
+import type { FC } from 'react'
+import { useStartLogin } from '@vtex/gatsby-theme-store/src/sdk/auth/useStartLogin'
+import { api } from '@vtex/gatsby-theme-store/src/sdk/auth/Service/api'
+
+const oAuthErrorFallbackUrl = () =>
+  new URL(api.oauth.error, window.location.origin).href
+
+const oAuthCallbackUrl = () =>
+  new URL(api.oauth.finish, window.location.origin).href
+
+const oAuthRedirectUrl = (providerName: string) => {
+  const search = new URLSearchParams()
+
+  search.append('providerName', providerName)
+  search.append('errorFallbackUrl', oAuthErrorFallbackUrl())
+
+  return `${api.pub.authentication.oauth.redirect}?${search.toString()}`
+}
+
+const ExternalProvider: FC<AuthProviderComponentProps> = ({
+  variant: v,
+  providerName = '',
+}) => {
+  const { formatMessage } = useIntl()
+  const startLogin = useStartLogin()
+  const [state, setState] = useState<string>('initial')
+  const variant = `customOAuth.${providerName}.${v}`
+
+  useEffect(() => {
+    ;(async () => {
+      if (!window) {
+        return
+      }
+
+      try {
+        await startLogin({
+          callbackUrl: oAuthCallbackUrl(),
+        })
+
+        window.location.href = oAuthRedirectUrl(providerName)
+      } catch {
+        setState('error')
+      }
+    })()
+  }, [providerName, startLogin, state])
+
+  return (
+    <Flex variant={variant}>
+      <Box variant={variant}>
```

Comment:
Is there a reason to put a Box inside a Flex using the same variant or is something missing that you intended to add?

Reply:
not really, didn't have time to go deeper in their implementation and never had actually used theme-ui so I guess I just copied that from somewhere else. I figured the flex would create a flexbox div and the box would just be a regular div, but I actually didn't confirm that 😅  if you have a better suggestion, I'm open to it :D Do you think just a box is enough?